plumbing/format/packfile/parser.go

```diff
@@ -248,7 +247,7 @@ func (p *Parser) parentReader(parent *ObjectHeader) (io.ReaderAt, error) {
 	// If the parent is not an external ref and we don't have the
 	// content offset, we won't be able to inflate via seeking through
 	// the packfile.
-	if parent.externalRef && parent.ContentOffset == 0 {
+	if !parent.externalRef && parent.ContentOffset == 0 {
```

Comment:
@pjbgf I found that we looking around this code... Don't know if the comment ("not an external") or the code ("parent.externalRef") was bad... I guess it was the code but no test fail in both case!?

Reply:
Good catch. We should expand on tests to avoid a future regression, but will merge this as-is to unblock you.

IMO, the change aligns [with the v5 code](https://github.com/go-git/go-git/blob/releases/v5.x/plumbing/format/packfile/parser.go#L402-L405) around thin packs.